File: 3838_Weighted_Word_Mapping/solution.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        r = []
        for w in words:
            s = 0
            for c in w:
                s += weights[ord(c) - ord("a")]
            r.append(chr(ord("z") - s % 26))
        return "".join(r)

File: 3838_Weighted_Word_Mapping/solution.c
char* mapWordWeights(char** words, int wordsSize, int* weights, int weightsSize) {
    char* r = (char*)malloc((wordsSize + 1) * sizeof(char));
    for (int i = 0; i < wordsSize; i++) {
        int s = 0;
        for (char* p = words[i]; *p; p++) s += weights[*p - 'a'];
        r[i] = 'z' - s % 26;
    }
    r[wordsSize] = '\0';
    return r;
}